core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -390,6 +391,30 @@ public SparkSqlDialect(SqlDialect.Context context) {
             timeUnitNode.getParserPosition());
         SqlFloorFunction.unparseDatetimeFunction(writer, call2, "DATE_TRUNC", false);
         break;
+      case COALESCE:
+        final SqlWriter.Frame coalesceFrame = writer.startFunCall("COALESCE");
```

Comment:
Please extract this into some function unparseCoalesce()

Reply:
done